ci-operator/step-registry/cucushift/installer/check/aws/custom-vpc/cucushift-installer-check-aws-custom-vpc-commands.sh

```diff
@@ -208,4 +208,115 @@ if [[ ${ASSIGN_ROLES_TO_SUBNETS} == "yes" ]]; then
   fi
 fi
 
+
+# ----------------------------------------------------------
+# Check if instances match CPMS spec
+# https://issues.redhat.com/browse/OCPBUGS-55492
+#
+# This check is applicable for all CAPI install*
+# For now, it will be enabled for 4.19+ only.*
+# We will enable the check on the previous versions depending on the backport
+# ----------------------------------------------------------
+
+all_equal() {
+  echo "Checking $*"
+  local first="$1"
+  shift
+  for s in "$@"; do
+    if [ "$first" != "$s" ]; then
+      return 1
+    fi
+  done
+  return 0
+}
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+  source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ocp_minor_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f2)
+ocp_major_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f1)
+
+if [ -f "${SHARED_DIR}/unset-proxy.sh" ] ; then
+  source "${SHARED_DIR}/unset-proxy.sh"
+fi
+
+if ((ocp_major_version == 4 && ocp_minor_version >= 19)); then
+
+  echo "Check if instances match CPMS spec ... "
+
+  workdir=$(mktemp -d)
+  out_machines=${workdir}/machines.json
+  out_instances=${workdir}/instances.json
+  out_subnets=${workdir}/subnets.json
+
+  if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
```

Comment:
duplicated codes with the lines 233 ~ 235 ?

Reply:
No, running `aws` command may fail behind a proxy,  so only enable proxy for `oc` command, then [unset proxy](https://github.com/openshift/release/pull/64695/files/494ba60e26bec5090fa1de93a5cad3eeebcdca61#diff-8fba81d123029dd084439f39df174ec7ed4c5d8211d4b7cfef0aa1542564eb80R240-R242)